### PR TITLE
Alter the folder for tekton pipelines & Templatization the tekton yamls

### PR DESCRIPTION
**Pull request by akash4sh**

### CODE REVIEW

File: cicd/tekton-pipelines/cluster-config/templates/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+      to use ssh-directory over basic-auth whenever possible and to bind a
+      Secret to this Workspace over other volume types.
+  - name: ssl-ca-directory
+    optional: true
+    description: |
+      A workspace containing CA certificates, this will be used by Git to
+      verify the peer with when fetching or pushing over HTTPS.
+  stepTemplate:
+    envFrom:
+    - secretRef:
```
Comment (indresh-28) [inline cicd/tekton-pipelines/cluster-config/templates/deploy.yaml]: Can we take these from values.yaml? that means we need to create a secret which has admin privilege using this name?

File: cicd/tekton-pipelines/cluster-config/templates/deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      fi
+      
+      export USERNAME=`cat /workspace/basic-auth/username`
+      git config --global user.email "${USER_EMAIL}"
+      git config --global user.name "${USERNAME}"  # Set your name here
+
+      git add .
+      git commit --allow-empty -m "updating imagetag to $(inputs.params.ImageTag)"
+
+      git push origin main
```
Comment (indresh-28) [inline cicd/tekton-pipelines/cluster-config/templates/deploy.yaml]: sometimes the repo has master also has default branch? can we check which is the default branch and use?

File: cicd/tekton-pipelines/cluster-config/templates/deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+  - name: wait-for-argocd-rollout
+    image: argoproj/argocd:v2.6.15
+    script: |
+      #!/usr/bin/env sh
+      set -e
+
+      argocd login $ARGOCD_SERVER --username $ARGOCD_USERNAME --password $ARGOCD_PASSWORD --insecure
```
Comment (indresh-28) [inline cicd/tekton-pipelines/cluster-config/templates/deploy.yaml]: Where this username and password should come from secret?

File: cicd/tekton-pipelines/cluster-config/templates/qt-test.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      Secret to this Workspace over other volume types.
+
+  stepTemplate:
+    env:
+    - name: QT_SERVER
+      value: {{ .Values.qualitytrace.server }}
+
+  steps:
+    - name: install-and-run-qt
+      image: ghcr.io/akash4sh/quality-trace:v2
```
Comment (indresh-28) [inline cicd/tekton-pipelines/cluster-config/templates/qt-test.yaml]: this image should come from kad or some org not from user repository

File: cicd/tekton-pipelines/cluster-config/values.yaml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+argocd:
+  server: argo.azureagent.optimizor.app
+  apps:
+    valuesPath: apps/vault-cred/values.yaml
```
Comment (indresh-28) [inline cicd/tekton-pipelines/cluster-config/values.yaml]: is this present in customer repository

File: cicd/tekton-pipelines/pipeline-template/values.yaml [hunk 6]
```diff
@@ -1,4 +1,6 @@
 pipelineName: "replace"
-ingressDomainName: "tekton.azureagent.optimizor.app"
+ingressDomainName: "tekton-event.azureagent.optimizor.app"
 tektonDashboard: "replace"
-namespace: "tekton-pipelines"
\ No newline at end of file
+namespace: "tekton-pipelines"
+configGitUrl: "gitops url"
+argoAppName: "appname"
```
Comment (indresh-28) [inline cicd/tekton-pipelines/pipeline-template/values.yaml:6]: which app name its referring here?

File: cicd/tekton-pipelines/pipeline-template/templates/pipeline.yaml [hunk 7]
```diff
@@ -73,6 +84,88 @@ spec:
...
+    - name: image
+      value: "$(params.imageUrl)@$(tasks.build.results.IMAGE_DIGEST)"
+    - name: eventname
+      value: "$(params.eventname)"
+    when:
+    - input: "$(params.eventname)"
+      operator: in
+      values: [ "push" , "tag" ,"release" ]
+
+  - name: cosign-image-verify
```
Comment (indresh-28) [inline cicd/tekton-pipelines/pipeline-template/templates/pipeline.yaml]: Is all these tasks tested?

File: cicd/tekton-pipelines/cluster-config/templates/deploy.yaml (hunk 3 above)
Comment (akash4sh) [reply cicd/tekton-pipelines/cluster-config/templates/deploy.yaml]: yes, its from secret

File: cicd/tekton-pipelines/pipeline-template/values.yaml (hunk 6 above)
Comment (akash4sh) [reply cicd/tekton-pipelines/pipeline-template/values.yaml:6]: application name which is need to be synced in argocd

File: cicd/tekton-pipelines/pipeline-template/templates/pipeline.yaml (hunk 7 above)
Comment (akash4sh) [reply cicd/tekton-pipelines/pipeline-template/templates/pipeline.yaml]: yes, tested

File: cicd/tekton-pipelines/cluster-config/templates/deploy.yaml (hunk 2 above)
Comment (akash4sh) [reply cicd/tekton-pipelines/cluster-config/templates/deploy.yaml]: or can we take this branch name also from values.yaml ?

Comment (indresh-28) [reply cicd/tekton-pipelines/cluster-config/templates/deploy.yaml]: depends on where we are pushing

**Pull request closed without merging**